### PR TITLE
resource control: GA for 7.1

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

as tittle

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/13732
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-resource-control.md [hunk 1]
```diff
@@ -33,6 +29,8 @@ With this feature, you can:
 
 - Combine multiple small and medium-sized applications from different systems into a single TiDB cluster. When the workload of an application grows larger, it does not affect the normal operation of other applications. When the system workload is low, busy applications can still be allocated the required system resources even if they exceed the set read and write quotas, so as to achieve the maximum utilization of resources.
 - Choose to combine all test environments into a single TiDB cluster, or group the batch tasks that consume more resources into a single resource group. It can improve hardware utilization and reduce operating costs while ensuring that critical applications can always get the necessary resources.
+- When there are various business workloads in the system, you can put different workloads into different resource groups. By using the resource control feature, you can ensure that the response time of transactional business is not affected by data analysis or batch business.
```
Comment (songrijie) [inline tidb-resource-control.md:32]:
```suggestion
- When there are mixed workloads in the same system, you can put different workloads into individual resource groups. By using the resource control feature, you can ensure that the response time of transactional applications is not affected by data analysis or batch applications.
```

File: tidb-resource-control.md [hunk 2]
```diff
@@ -33,6 +29,8 @@ With this feature, you can:
...
+- When the cluster encounters an unexpected SQL performance problem, you can use SQL Binding and resource groups to temporarily limit the resource consumption of a SQL statement.
```
Comment (songrijie) [inline tidb-resource-control.md:33]:
```suggestion
- When the cluster encounters an unexpected SQL performance problem, you can use SQL Binding along with resource groups to temporarily limit the resource consumption of a SQL statement.
```

File: tidb-resource-control.md [hunk 3]
```diff
@@ -223,7 +232,21 @@ TiKV also records the request QPS from different resource groups in Grafana's **
...
+
+## FAQ
+
+1. If I do not want to use resource groups to control resources for now, do I have to disable this feature?
+
+    No. Users who do not specify any resource groups will be placed in the `default` resource group that has unlimited resources. When all users belong to the `default` resource group, the resource allocation method is the same as when the resource control is disabled.
+
+2. Can a database user be bound to several resource groups?
+
+    No. A database user can only be bound to one resource group. However, during the session runtime, you can use [`SET RESOURCE GROUP`](/sql-statements/sql-statement-set-resource-group.md) to set the resource group used by the current session. You can also use the optimizer [`RESOURCE_GROUP()`](/optimizer-hints.md#resource_groupresource_group_name) Hint to set the resource group for the running statement.
```
Comment (songrijie) [inline tidb-resource-control.md:245]:
```suggestion
    No. A database user can only be bound to one resource group. However, during the session runtime, you can use [`SET RESOURCE GROUP`](/sql-statements/sql-statement-set-resource-group.md) to set the resource group used by the current session. You can also use the optimizer hint [`RESOURCE_GROUP()`](/optimizer-hints.md#resource_groupresource_group_name) to set the resource group for the running statement.
```

File: tidb-resource-control.md [hunk 4]
```diff
@@ -223,7 +232,21 @@ TiKV also records the request QPS from different resource groups in Grafana's **
...
+
+3. What happens when the total usage (`RU_PER_SEC`) of all resource groups exceeds the system capacity?
```
Comment (songrijie) [inline tidb-resource-control.md:247]:
```suggestion
3. What happens when the total resource allocation (`RU_PER_SEC`) of all resource groups exceeds the system capacity?
```

File: tidb-resource-control.md [hunk 5]
```diff
@@ -223,7 +232,21 @@ TiKV also records the request QPS from different resource groups in Grafana's **
...
+
+    TiDB does not check the capacity when you create a resource group. As long as the system has enough available resources, TiDB can meet the usage requirements of each resource group. When the system resources exceed the limit, TiDB will first meet the requests of the high-priority (PRIORITY) resource groups. If the requests of the same priority cannot be met, TiDB will allocate resources proportionally according to the usage (`RU_PER_SEC`).
```
Comment (songrijie) [inline tidb-resource-control.md:249]:
```suggestion
    TiDB does not verify the capacity when you create a resource group. As long as the system has enough available resources, TiDB can meet the resource requirements of each resource group. When the system is saturated, TiDB will first meet the requests coming from resource groups with high priority. If the requests with the same priority cannot be met, TiDB will allocate resources proportionally according to the resource allocation (`RU_PER_SEC`).
```

File: tidb-resource-control.md [hunk 6]
```diff
@@ -90,6 +88,17 @@ For more information about the resource control mechanism and parameters, see [R
 
 ## How to use resource control
 
+This section describes how to use the resource control feature to manage resource groups and control the resource usage of each resource group.
```
Comment (HuSharp) [inline tidb-resource-control.md:91]:
```suggestion
This section describes how to use the resource control feature to manage resource groups and control the resource allocation of each resource group.
```

File: tidb-resource-control.md [hunk 7]
```diff
@@ -33,6 +29,8 @@ With this feature, you can:
 
 - Combine multiple small and medium-sized applications from different systems into a single TiDB cluster. When the workload of an application grows larger, it does not affect the normal operation of other applications. When the system workload is low, busy applications can still be allocated the required system resources even if they exceed the set read and write quotas, so as to achieve the maximum utilization of resources.
 - Choose to combine all test environments into a single TiDB cluster, or group the batch tasks that consume more resources into a single resource group. It can improve hardware utilization and reduce operating costs while ensuring that critical applications can always get the necessary resources.
+- When there are mixed workloads in the same system, you can put different workloads into individual resource groups. By using the resource control feature, you can ensure that the response time of transactional applications is not affected by data analysis or batch applications.
```
Comment (hfxsd) [inline tidb-resource-control.md:32]:
```suggestion
- When there are mixed workloads in the same system, you can put different workloads into separate resource groups. By using the resource control feature, you can ensure that the response time of transactional applications is not affected by data analysis or batch applications.
```

**Review by songrijie — APPROVED** [review]
LGTM

**Review by HuSharp — APPROVED** [review]
LGTM! may need to wait for https://github.com/pingcap/docs/pull/13270 to merge before passing the CI.

File: tidb-resource-control.md [hunk 8]
```diff
@@ -33,6 +29,8 @@ With this feature, you can:
 
 - Combine multiple small and medium-sized applications from different systems into a single TiDB cluster. When the workload of an application grows larger, it does not affect the normal operation of other applications. When the system workload is low, busy applications can still be allocated the required system resources even if they exceed the set read and write quotas, so as to achieve the maximum utilization of resources.
 - Choose to combine all test environments into a single TiDB cluster, or group the batch tasks that consume more resources into a single resource group. It can improve hardware utilization and reduce operating costs while ensuring that critical applications can always get the necessary resources.
+- When there are mixed workloads in the same system, you can put different workloads into separate resource groups. By using the resource control feature, you can ensure that the response time of transactional applications is not affected by data analysis or batch applications.
+- When the cluster encounters an unexpected SQL performance problem, you can use SQL Binding along with resource groups to temporarily limit the resource consumption of a SQL statement.
```
Comment (Oreoxmt) [inline tidb-resource-control.md:33]:
```suggestion
- When there are mixed workloads in a system, you can put different workloads into separate resource groups. By using the resource control feature, you can ensure that the response time of transactional applications is not affected by data analysis or batch applications.
- When the cluster encounters an unexpected SQL performance issue, you can use SQL bindings along with resource groups to temporarily limit the resource consumption of a SQL statement.
```

File: tidb-resource-control.md [hunk 9]
```diff
@@ -90,6 +88,17 @@ For more information about the resource control mechanism and parameters, see [R
 
 ## How to use resource control
 
+This section describes how to use the resource control feature to manage resource groups and control the resource allocation of each resource group.
+
+### Methods for estimating capacity
```
Comment (Oreoxmt) [inline tidb-resource-control.md:93]:
```suggestion
### Estimate cluster capacity
```

File: tidb-resource-control.md [hunk 10]
```diff
@@ -90,6 +88,17 @@ For more information about the resource control mechanism and parameters, see [R
...
+
+Before resource planning, you need to know the overall capacity of the cluster. TiDB provides the statement [`CALIBRATE RESOURCE`](/sql-statements/sql-statement-calibrate-resource.md) to estimate the cluster capacity. TiDB provides two estimation methods:
```
Comment (Oreoxmt) [inline tidb-resource-control.md:95]:
```suggestion
Before resource planning, you need to know the overall capacity of the cluster. TiDB provides the statement [`CALIBRATE RESOURCE`](/sql-statements/sql-statement-calibrate-resource.md) to estimate the cluster capacity. You can use one of the following methods:
```

File: tidb-resource-control.md [hunk 11]
```diff
@@ -223,7 +232,21 @@ TiKV also records the request QPS from different resource groups in Grafana's **
...
 ## Tool compatibility
 
-The resource control feature is still in its experimental stage. It does not impact the regular usage of data import, export, and other replication tools. BR, TiDB Lightning, and TiCDC do not currently support processing DDL operations related to resource control, and their resource consumption is not limited by resource control.
+The resource control feature does not impact the regular usage of data import, export, and other replication tools. BR, TiDB Lightning, and TiCDC do not currently support processing DDL operations related to resource control, and their resource consumption is not limited by resource control.
+
+## FAQ
+
+1. If I do not want to use resource groups to control resources for now, do I have to disable this feature?
+
+    No. Users who do not specify any resource groups will be placed in the `default` resource group that has unlimited resources. When all users belong to the `default` resource group, the resource allocation method is the same as when the resource control is disabled.
```
Comment (Oreoxmt) [inline tidb-resource-control.md:241]:
```suggestion
    No. Users who do not specify any resource groups will be bound to the `default` resource group that has unlimited resources. When all users belong to the `default` resource group, the resource allocation method is the same as when the resource control is disabled.
```

File: tidb-resource-control.md [hunk 12]
```diff
@@ -223,7 +232,21 @@ TiKV also records the request QPS from different resource groups in Grafana's **
 
 ## Tool compatibility
 
-The resource control feature is still in its experimental stage. It does not impact the regular usage of data import, export, and other replication tools. BR, TiDB Lightning, and TiCDC do not currently support processing DDL operations related to resource control, and their resource consumption is not limited by resource control.
+The resource control feature does not impact the regular usage of data import, export, and other replication tools. BR, TiDB Lightning, and TiCDC do not currently support processing DDL operations related to resource control, and their resource consumption is not limited by resource control.
+
+## FAQ
+
+1. If I do not want to use resource groups to control resources for now, do I have to disable this feature?
```
Comment (Oreoxmt) [inline tidb-resource-control.md:239]:
```suggestion
1. Do I have to disable resource control if I don't want to use resource groups?
```

File: tidb-resource-control.md [hunk 13]
```diff
@@ -223,7 +232,21 @@ TiKV also records the request QPS from different resource groups in Grafana's **
...
+
+2. Can a database user be bound to several resource groups?
+
+    No. A database user can only be bound to one resource group. However, during the session runtime, you can use [`SET RESOURCE GROUP`](/sql-statements/sql-statement-set-resource-group.md) to set the resource group used by the current session. You can also use the optimizer hint [`RESOURCE_GROUP()`](/optimizer-hints.md#resource_groupresource_group_name) to set the resource group for the running statement.
+
+3. What happens when the total resource allocation (`RU_PER_SEC`) of all resource groups exceeds the system capacity?
+
+    TiDB does not verify the capacity when you create a resource group. As long as the system has enough available resources, TiDB can meet the resource requirements of each resource group. When the system is saturated, TiDB will first meet the requests coming from resource groups with higher priority. If the requests with the same priority cannot be met, TiDB will allocate resources proportionally according to the resource allocation (`RU_PER_SEC`).
```
Comment (Oreoxmt) [inline tidb-resource-control.md:249]:
```suggestion
    TiDB does not verify the capacity when you create a resource group. As long as the system has enough available resources, TiDB can meet the resource requirements of each resource group. When the system resources exceed the limit, TiDB prioritizes satisfying requests from resource groups with higher priority. If requests with the same priority cannot all be met, TiDB allocates resources proportionally according to the resource allocation (`RU_PER_SEC`).
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**